### ISSUE TITLE
aws s3 with kms : encryption context value not working as expected

### ISSUE THREAD

**Issue by imdhruva** [thanos-io/thanos#4138]
**Thanos, Prometheus and Golang version used**:

```
thanos, version 0.20.0 (branch: HEAD, revision: 893188a6752e260cf54f3c082392a521b6c92f75)
  build user:       circleci@c7e18ff0c722
  build date:       20210428-13:07:30
  go version:       go1.16.3
  platform:         linux/amd64
```
**Object Storage Provider**:
AWS

**What happened**:
1. Upgrade thanos image from `v0.18.0` to `v0.20.0`
2. Errors observed in `thanos-sidecar` :
```
level=warn ts=2021-05-03T07:23:48.984793685Z caller=sidecar.go:291 err="upload 01F4DS38ZY7NYQNP0J7Q48DK1N: upload debug meta file: upload s3 object: The context value for 'aws:s3:arn' shall be equal to object's ARN" uploaded=0
```
**What you expected to happen**:
Since the config has not changed the error above should not happen

**How to reproduce it (as minimally and precisely as possible)**:
1. storate config 
 ```
  type: yaml
  value:
    type: S3
    config:
      bucket: <bucket-name>
      endpoint: s3.<aws-region >.amazonaws.com
      region: <aws-region >
      signature_version2: false
      insecure: false
      sse_config:
        type: SSE-KMS
        kms_key_id: arn:aws:kms:<aws-region >:<aws-account-id>:key/<kms-key-id >
        kms_encryption_context:
          "aws:s3:arn": "arn:aws:s3:::<bucket-name>"
```
**Full logs to relevant components**:
   ```
level=warn ts=2021-05-03T07:23:48.984793685Z caller=sidecar.go:291 err="upload 01F4DS38ZY7NYQNP0J7Q48DK1N: upload debug meta file: upload s3 object: The context value for 'aws:s3:arn' shall be equal to object's ARN" uploaded=0
  ```

**Anything else we need to know**:

**Comment by wiardvanrij** [comment]
Hi thanks for the report. I went over the changes for S3 and from 0.18 - 0.20 I don't see anything relevant that has been changed. 

To be certain that it's related to Thanos; Could you perhaps test if reverting back to 0.18 fixes your issue? If so, please try Thanos 0.19. If this does not resolve your issue, could it be that there is something changed in AWS / configuration? 

Last thing, this might be completely irrelevant but you have: 
```
       kms_encryption_context:
         "aws:s3:arn": "arn:aws:s3:::<bucket-name>"
```

I haven't tested this, but the docs state:
```
kms_encryption_context: {}
``` 
Which lacks the `"` on the `kms_encryption_context` - Perhaps you could try the setup used as example?

**Comment by imdhruva** [comment]
@wiardvanrij this was tried already. the error re-occurs for `v0.19`. It disappears after rolling back to `v0.18`. Also, to confirm the configuration has not changed nor has the attached IAM Role/Policies.
